### PR TITLE
created local reactnatice sortable list example

### PR DESCRIPTION
**Pull request by cmreddy071**
Created react native sortable-list GitHub example.

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -1,19 +1,96 @@
...
+  },
+];
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { data: data, sortable: false };
+  }
+
+  renderDragableItem = ({ item, index, move, moveEnd, isActive }) => (
+    <TouchableOpacity onLongPress={move} onPressOut={moveEnd}>
```
Comment (joebochill) [inline App.js]: Can we configure the longPress duration? Right now it feels like too long you have to press before you can drag. Compare to e.g. the mail app on your phone where you can drag with a regular press. Maybe just use onPressIn?

File: App.js [hunk 2]
```diff
@@ -1,19 +1,96 @@
...
+  };
+
+  render() {
+    return (
+      <View style={styles.container}>
+        <Header
+          backgroundColor={Colors.blue[500]}
+          centerComponent={{
+            text: 'Sortable List',
+            style: { color: Colors.white[500], fontSize: 16 },
```
Comment (joebochill) [inline App.js]: isn't pure white white[50]?

File: App.js [hunk 3]
```diff
@@ -1,19 +1,96 @@
...
+          )}
+        />
+        <DraggableFlatList
+          data={this.state.data}
+          renderItem={
+            this.state.sortable ? this.renderDragableItem : this.renderItem
+          }
+          keyExtractor={(item, index) => `${index}-${this.state.sortable}`}
+          scrollPercent={50}
+          onMoveEnd={({ data }) => this.setState({ data })}
```
Comment (joebochill) [inline App.js]: what do these lines do?

File: package.json [hunk 4]
```diff
@@ -9,19 +9,27 @@
...
   "dependencies": {
+    "@pxblue/colors": "^1.0.13",
     "expo": "^35.0.0",
     "react": "16.8.3",
     "react-dom": "16.8.3",
     "react-native": "https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz",
-    "react-native-web": "^0.11.7",
-    "@pxblue/colors": "^1.0.13"
+    "react-native-elements": "^1.2.0",
+    "react-native-gesture-handler": "^1.4.1",
```
Comment (joebochill) [inline package.json]: when running the application, there's an error in the console about incompatible versions with this package

File: tests/App.test.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+
+import App from '../App';
+
+describe('App Tests ', function () {
```
Comment (joebochill) [inline tests/App.test.js]: Can we add any tests that test the functionality of the sort? Or at least test that the list is not sortable if the edit button hasn't been clicked?

Comment (cmreddy071) [reply tests/App.test.js]: Added new tests

File: package.json (hunk 4 above)
Comment (cmreddy071) [reply package.json]: fixed the error in console.

File: App.js (hunk 3 above)
Comment (cmreddy071) [reply App.js]: This came with reactnative-draggable-flatlist.

File: App.js (hunk 2 above)
Comment (cmreddy071) [reply App.js]: changed to pure[50].

File: App.js (hunk 1 above)
Comment (cmreddy071) [reply App.js]: Long press duration changed.

File: App.js [hunk 6]
```diff
@@ -63,26 +64,36 @@ export default class App extends React.Component {
...
-          onMoveEnd={({ data }) => this.setState({ data })}
-        />
+        {
+          this.state.sortable
+            ? (
+              <DraggableFlatList
+                data={this.state.data}
+                renderItem={this.renderDragableItem}
+                keyExtractor={(item, index) => `${index}`}
+                onMoveEnd={({ data }) => this.setState({ data })}
```
Comment (joebochill) [inline App.js]: What does onMoveEnd do...if you're writing the code, you should know what it does.

File: App.js [hunk 7]
```diff
@@ -91,6 +102,6 @@ export default class App extends React.Component {
 const styles = StyleSheet.create({
   container: {
     flex: 1,
-    backgroundColor: Colors.white[100],
+    backgroundColor: Colors.white[50],
```
Comment (joebochill) [inline App.js]: page background is supposed to be gray...pure white is for the text

File: package.json [hunk 8]
```diff
@@ -10,22 +10,24 @@
...
     "react-dom": "16.8.3",
     "react-native": "https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz",
+    "react-native-draggable-flatlist": "1.1.9",
     "react-native-elements": "^1.2.0",
-    "react-native-gesture-handler": "^1.4.1",
-    "react-native-web": "^0.11.4",
+    "react-native-gesture-handler": "~1.3.0",
     "react-native-modal": "^11.3.1",
-    "react-native-draggable-flatlist": "1.1.9",
+    "react-native-web": "^0.11.4",
```
Comment (joebochill) [inline package.json]: why are we adding this?

File: App.js (hunk 6 above)
Comment (cmreddy071) [reply App.js]: This will keep new ordered data in state once the drag end.

File: package.json (hunk 8 above)
Comment (cmreddy071) [reply package.json]: Its not the newly added one. It was there in reactnative branch. Hope it was installed by expo.

File: App.js (hunk 7 above)
Comment (cmreddy071) [reply App.js]: I cross checked with other examples, everywhere we are using white color only for background.

**Comment by cmreddy071** [comment]
> * The list item height should not change between edit/non-edit modes
Icon size reduced to match with list item height in edit mode.

**Pull request merged**